### PR TITLE
Add `Rpc` rpc modules

### LINKED ISSUE #3972: Rpc_modules not supported for geth attach command
**Describe the bug**
Contact @ethdreamer for more details (Lighthouse team)

**To Reproduce**
Geth attach on Nethermind

**Expected behavior**
Would work

### PR DESCRIPTION
**Pull request by asdacap**
Closes #3972

Replicate geth's `rpc_modules` command. so that `geth attach` would work.

## Changes:
- Add `RpcRpcModule`.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing

- Call method, got
```
{'jsonrpc': '2.0', 'result': {'eth': '1.0', 'proof': '1.0', 'debug': '1.0', 'trace': '1.0', 'personal': '1.0', 'txPool': '1.0', 'net': '1.0', 'parity': '1.0', 'subscribe': '1.0', 'web3': '1.0'}, 'id': 12}
```
- Specify ipc port to `~/.ethereum/geth.ipc`, run `get attach`, call `eth.getBlockByNumber(1)`, seems to return a block.

## Further comments (optional)

`Rpc` module is not turned on by default on most config.

### CODE REVIEW

File: src/Nethermind/Nethermind.Init/Steps/RegisterRpcModules.cs [hunk 1]
```diff
@@ -231,6 +232,9 @@ public virtual async Task Execute(CancellationToken cancellationToken)
                 await plugin.InitRpcModules();
             }
             
+            RpcRpcModule rpcRpcModule = new(rpcModuleProvider.Enabled.ToList());
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Init/Steps/RegisterRpcModules.cs:235]: No need to ToList(), jusst pass IReadOnlyCollection

File: src/Nethermind/Nethermind.JsonRpc/Modules/Module/RpcRpcModule.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    public RpcRpcModule(List<String> enabledModules)
+    {
+        _enabledModules = enabledModules;
+    }
+    
+    public ResultWrapper<IDictionary<string, string>> rpc_modules()
+    {
+        // Geth seems to fix version at 1.0t 
+        IDictionary<string, string> response = _enabledModules.ToDictionary((s => s), s => "1.0");
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc/Modules/Module/RpcRpcModule.cs:38]: Not sure if needs to be recreated every call or just precomputed? Potentially if we want to enable/disable modules on the fly, but even then we can use some kind of notification mechanism

File: src/Nethermind/Nethermind.JsonRpc/Modules/Module/RpcRpcModule.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+public class RpcRpcModule: IRpcRpcModule
+{
+    private readonly List<String> _enabledModules;
+
+    public RpcRpcModule(List<String> enabledModules)
+    {
+        _enabledModules = enabledModules;
+    }
+    
+    public ResultWrapper<IDictionary<string, string>> rpc_modules()
```
[the remark this answers is not in the document]
Comment (asdacap) [reply src/Nethermind/Nethermind.JsonRpc/Modules/Module/RpcRpcModule.cs:35]: Which one is that, she small one or the large one?

[the remark this answers is not in the document]
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.JsonRpc/Modules/Module/RpcRpcModule.cs:35]: Small, large is type name

File: src/Nethermind/Nethermind.Runner/Properties/launchSettings.json [hunk 4]
```diff
@@ -180,7 +180,7 @@
     },
     "Kiln": {
       "commandName": "Project",
-      "commandLineArgs": "--config kiln.cfg  --Init.DiagnosticMode MemDb --Merge.TerminalBlockNumber null"
+      "commandLineArgs": "--config kiln.cfg --baseDbPath /home/amirul/nethermind_kiln --JsonRpc.IpcUnixDomainSocketPath /home/amirul/.ethereum/geth.ipc"
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/Properties/launchSettings.json:183]: please remove changes

File: src/Nethermind/Nethermind.Runner/Properties/launchSettings.json [hunk 5]
```diff
@@ -200,7 +200,10 @@
     },
     "Ropsten Archive": {
       "commandName": "Project",
-      "commandLineArgs": "--config configs\\ropsten_archive.cfg --baseDbPath C:\\nethermind_db"
+      "commandLineArgs": "--config configs\\ropsten_archive.cfg --baseDbPath C:\\nethermind_db",
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/Properties/launchSettings.json:203]: please remove changes

File: src/Nethermind/Nethermind.Runner/configs/kiln.cfg [hunk 6]
```diff
@@ -21,7 +21,8 @@
     "Timeout": 20000,
     "Host": "127.0.0.1",
     "Port": 8545,
-    "EnabledModules": ["Eth", "Subscribe", "Trace", "TxPool", "Web3", "Personal", "Proof", "Net", "Parity", "Health"],
+    "JwtSecretFile": "/tmp/jwtsecret",
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/configs/kiln.cfg:24]: please remove changes

**Review by LukaszRozmej — CHANGES_REQUESTED** [review]
Remove config changes and submodule changes, otherwise we can merge

**Pull request merged**